onnx_tf/handlers/backend/tfidf_vectorizer.py

```diff
@@ -0,0 +1,123 @@
+import numpy as np
+import tensorflow as tf
+
+from onnx_tf.handlers.backend_handler import BackendHandler
+from onnx_tf.handlers.handler import onnx_op
+from onnx_tf.handlers.handler import tf_func
+
+
+@onnx_op("TfIdfVectorizer")
+class TfIdfVectorizer(BackendHandler):
+
+  @classmethod
+  def args_check(cls, node, **kwargs):
+    if "pool_int64s" in node.attrs and "pool_strings" in node.attrs:
+      raise ValueError(
+          "Cannot set the pool_int64s and pool_strings in an input at the same time."
+      )
+
+  @classmethod
+  def _prepare_ngrams(cls, x, n, skip):
+    # This method transform input into n-grams for a specific skip
+    # input x is a 1D tensor
+    # ex1: x=[1,2,3,4] n=1 skip=0 output=[[1],[2],[3],[4]]
+    # ex2: x=[1,2,3,4] n=2 skip=0 output=[[1,2],[2,3],[3,4]]
+    # ex3: x=[1,2,3,4] n=2 skip=1 output=[[1,3],[2,4]]
+    count = x.shape[0] - n + 1 - skip
+    mul = skip + 1
+    ngrams = [x[i * mul:i * mul + count] for i in range(n)]
+    ngrams = tf.stack(ngrams)
+    ngrams = tf.transpose(ngrams, [1, 0])
+    return ngrams
+
+  @classmethod
+  def _calc_ngram(cls, x, pool, n, max_skip):
+    # This method calculates ngram counts for a specific n and
+    # all allowable skips
+
+    # For 1gram, skip is not in use
+    if n == 1:
+      return cls._calc_ngram_skip(x, pool, n)
+
+    # Loop through maximum allowable skip count and sum up the results
+    result = tf.zeros([int(len(pool) / n)], dtype=tf.int32)
+    max_allowable_skip = np.minimum(max_skip,
+                                    int((int(x.shape[0]) - 1) / (n - 1) - 1))
+    for skip in range(max_allowable_skip + 1):
+      # For each skip calculate the ngram counts
+      result += cls._calc_ngram_skip(x, pool, n, skip)
+
+    return result
+
+  @classmethod
+  def _calc_ngram_skip(cls, x, pool, n, skip=0):
+    # This method calculates ngram counts for specific n and skip
+
+    # Make pool into an array of ngrams
+    pool = np.reshape(pool, (int(len(pool) / n), n))
+
+    # Make input as an array of ngrams
+    new_x = cls._prepare_ngrams(x, n, skip)
+
+    # Loop through the ngram targets in the pool
+    for i in range(len(pool)):
+      ngram_count = tf.map_fn(lambda in_x: tf.where(tf.reduce_all(tf.equal(in_x, tf.constant(pool[i], dtype=new_x.dtype))), tf.constant([1]), tf.constant([0])), new_x, dtype=tf.int32)
+      ngram_count = tf.math.count_nonzero(ngram_count, dtype=tf.int32)
+      ngram_count = tf.reshape(ngram_count, [1])
+      # Building up the result
+      if i == 0:
```

Comment:
Can we store intermediate results in a tensor list and do a single concatenation in the end? Feels like this can potentially reduce the number of concat nodes in our imported graph.

Reply:
A tensor_list is used to minimize the number of concat nodes.